wallet-new/src/Cardano/Wallet/Server/Plugins.hs

```diff
@@ -163,3 +160,35 @@ updateNotifier :: Plugin Kernel.WalletMode
 updateNotifier = [
     \_diffusion -> logError "Not Implemented: updateNotifier [CBR-374]"
     ]
+
+-- | A @Middleware@ to throttle requests.
+throttleMiddleware :: Maybe ThrottleSettings -> Middleware
+throttleMiddleware Nothing app = app
+throttleMiddleware (Just ts) app = \req respond -> do
+    throttler <- Throttle.initThrottler
+    Throttle.throttle throttleSettings throttler app req respond
+  where
+    throttleSettings = Throttle.defaultThrottleSettings
+        { Throttle.onThrottled = \microsTilRetry ->
+            let
+                err = V1.RequestThrottled microsTilRetry
+            in
+                responseLBS (V1.toHttpErrorStatus err) [applicationJson] (encode err)
```

Comment:
I have used V1 WalletErrors to report error here but I am not actually sure is this the right way to report errors with new data layer?